File: Xamarin_Forms/Xamarin_Forms/GertecXamarinForms/GertecXamarinForms.Android/ExemploImpressora/ConfigPrint.cs
using Xamarin.Forms;

namespace GertecXamarinForms.Droid.ExemploImpressora
{
    public class ConfigPrint
    {
        private string fonte = "NORMAL";
        private string alinhamento;
        private int tamanho;
        private int offSet;
        private int iHeight;
        private int lineSpace;
        private bool negrito;
        private bool italico;
        private bool sublinhado;
        private int avancaLinhas;

        public ConfigPrint()
        {
            this.fonte = "NORMAL";
            this.alinhamento = "CENTER";
            this.tamanho = 20;
            this.offSet = 0;
            this.iHeight = 400;
            this.IWidth = 300;
            this.lineSpace = 0;
            this.negrito = true;
            this.italico = true;
            this.sublinhado = false;
            this.avancaLinhas = 10;
        }

        public ConfigPrint(string fonte,
                           string alinhamento,
                           int tamanho,
                           int offSet,
                           int lineSpace,
                           bool negrito,
                           bool italico,
                           bool sublinhado)
        {
            this.fonte = fonte;
            this.alinhamento = alinhamento;
            this.tamanho = tamanho;
            this.offSet = offSet;
            this.lineSpace = lineSpace;
            this.negrito = negrito;
            this.italico = italico;
            this.sublinhado = sublinhado;
        }

        public ConfigPrint(string fonte,
                           string alinhamento,
                           int tamanho,
                           int offSet,
                           int iHeight,
                           int iWidth,
                           int lineSpace,
                           bool negrito,
                           bool italico,
                           bool sublinhado)
        {
            this.fonte = fonte;
            this.alinhamento = alinhamento;
            this.tamanho = tamanho;
            this.offSet = offSet;
            this.iHeight = iHeight;
            this.IWidth = iWidth;
            this.lineSpace = lineSpace;
            this.negrito = negrito;
            this.italico = italico;
            this.sublinhado = sublinhado;
        }

        public string Fonte
        {
            get { return this.fonte; }
            set { this.fonte = value; }
        }

        public string Alinhamento
        {
            get { return this.alinhamento; }
            set { this.alinhamento = value; }
        }

        public int Tamanho
        {
            get { return this.tamanho; }
            set { this.tamanho = value; }
        }

        public int OffSet
        {
            get { return this.offSet; }
            set { this.offSet = value; }
        }

        public int IHeight
        {
            get { return this.iHeight; }
            set { this.iHeight = value; }
        }

        public int IWidth { get; set; }

        public int LineSpace
        {
            get { return this.lineSpace; }
            set { this.lineSpace = value; }
        }

        public bool Negrito
        {
            get { return this.negrito; }
            set { this.negrito = value; }
        }

        public bool Italico
        {
            get { return this.italico; }
            set { this.italico = value; }
        }

        public bool SubLinhado
        {
            get { return this.sublinhado; }
            set { this.sublinhado = value; }
        }

        public int AvancaLinha
        {
            get { return this.avancaLinhas; }
            set { this.avancaLinhas = value; }
        }
    }
}

File: Flutter/Flutter/android/app/src/main/java/com/example/flutter_gertec/ConfigPrint.java
package com.example.flutter_gertec;

public class ConfigPrint {

    private String fonte = "NORMAL";
    private String alinhamento;
    private int tamanho;
    private int offSet;
    private int iHeight;
    private int iWidth;
    private int lineSpace;
    private boolean negrito;
    private boolean italico;
    private boolean sublinhado;
    private int avancaLinhas;

    public ConfigPrint() {
        this.fonte = "NORMAL";
        this.alinhamento = "CENTER";
        this.tamanho = 20;
        this.offSet = 0;
        this.iHeight = 700;
        this.iWidth = 430;
        this.lineSpace = 0;
        this.negrito = true;
        this.italico = true;
        this.sublinhado = false;
        this.avancaLinhas = 0;
    }

    public ConfigPrint(String fonte, String alinhamento, int tamanho, int offSet, int lineSpace, boolean negrito,
            boolean italico, boolean sublinhado) {
        this.fonte = fonte;
        this.alinhamento = alinhamento;
        this.tamanho = tamanho;
        this.offSet = offSet;
        this.lineSpace = lineSpace;
        this.negrito = negrito;
        this.italico = italico;
        this.sublinhado = sublinhado;
    }

    public ConfigPrint(String fonte, String alinhamento, int tamanho, int offSet, int iHeight, int iWidth,
            int lineSpace, boolean negrito, boolean italico, boolean sublinhado) {
        this.fonte = fonte;
        this.alinhamento = alinhamento;
        this.tamanho = tamanho;
        this.offSet = offSet;
        this.iHeight = iHeight;
        this.iWidth = iWidth;
        this.lineSpace = lineSpace;
        this.negrito = negrito;
        this.italico = italico;
        this.sublinhado = sublinhado;
    }

    public String getFonte() {
        return fonte;
    }

    public void setFonte(String fonte) {
        this.fonte = fonte;
        switch (fonte) {
            case "NORMAL":
                break;
            case "DEFAULT":
                break;
            case "DEFAULT BOLD":
                break;
            case "MONOSPACE":
                break;
            case "SANS SERIF":
                break;
            case "SERIF":
                break;
            default:
                setFont(fonte);
        }
    }

    public String getAlinhamento() {
        return alinhamento;
    }

    public void setAlinhamento(String alinhamento) {
        this.alinhamento = alinhamento;
    }

    public int getTamanho() {
        return tamanho;
    }

    public void setTamanho(int tamanho) {
        this.tamanho = tamanho;
    }

    public int getOffSet() {
        return offSet;
    }

    public void setOffSet(int offSet) {
        this.offSet = offSet;
    }

    public int getLineSpace() {
        return lineSpace;
    }

    public void setLineSpace(int lineSpace) {
        this.lineSpace = lineSpace;
    }

    private void setFont(String fonte) {
        this.fonte = "fonts/" + fonte;
    }

    public int getiHeight() {
        return iHeight;
    }

    public void setiHeight(int iHeight) {
        this.iHeight = iHeight;
    }

    public int getiWidth() {
        return iWidth;
    }

    public void setiWidth(int iWidth) {
        this.iWidth = iWidth;
    }

    public boolean isNegrito() {
        return negrito;
    }

    public void setNegrito(boolean negrito) {
        this.negrito = negrito;
    }

    public boolean isItalico() {
        return italico;
    }

    public void setItalico(boolean italico) {
        this.italico = italico;
    }

    public boolean isSublinhado() {
        return sublinhado;
    }

    public void setSublinhado(boolean sublinhado) {
        this.sublinhado = sublinhado;
    }

    public int getAvancaLinhas() {
        return avancaLinhas;
    }

    public void setAvancaLinhas(int avancaLinhas) {
        this.avancaLinhas = avancaLinhas;
    }
}
